File: backend/backend/DTO/UserDto.cs
using System.Text.Json.Serialization;

namespace backend.DTO;
public class UserDto
{
    public required Guid Id { get; set; }
    public required string Firstname { get; set; }
    public required string Lastname { get; set; }
    public required string Email { get; set; }
    public required string PhoneNumber { get; set; }
    public required bool IdentifiedWithGoogle { get; set; }
}

public class ConnectedUserDto : UserDto
{
    public string Token { get; set; } = "";
}

public class AuthUserDto(string email, string password)
{
    public string Email { get; set; } = email;
    public string Password { get; set; } = password;
}

public class CreateUserDto
{
    public required string Firstname { get; set; }
    public required string Lastname { get; set; }
    public required string Email { get; set; }
    public required string PhoneNumber { get; set; }
    public required string Password { get; set; }
}

public class GoogleUserDto
{
    public required string Email { get; set; }
    public string Lastname { get; set; } = string.Empty;
    public string Firstname { get; set; } = string.Empty;
}

File: frontend/src/app/DTO/UserDto.ts
export interface AuthUserDto {
    email: string,
    password: string,
}

export interface CreateUserDto {
    firstname: string,
    lastname: string,
    email: string,
    phoneNumber: string,
    password: string,
}

export interface UserDto {
    id: string,
    firstname: string,
    lastname: string,
    email: string,
    phoneNumber: string
}

export interface ConnectedUserDto extends UserDto {
    token: string,
}